src/model/NPMResolver.ts

```diff
@@ -323,4 +325,48 @@ export class NPMResolver {
     private getPackageLockPath(): string {
         return path.resolve(this.mainRepo, NPMResolver.PACKAGE_LOCK_JSON);
     }
+
+    private initRegistry(): void {
+        console.info("⟲ Initialising cplace jfrog registry for NPM");
+        const registry = '//cplace.jfrog.io/artifactory/api/npm/cplace-npm-local/';
+        const gradleProps: string = (execSync(path.join(this.mainRepo, 'gradlew properties')) || '').toString();
+
+        if (!gradleProps) {
+            console.error('✗ Gradle Properties not found');
+        } else {
+            const token: string | undefined = (gradleProps.match(/repo\.cplace\.apiToken: *([a-z0-9]+)/gi) || [])[0];
+            const user: string | undefined = (gradleProps.match(/repo\.cplace\.apiTokenUser: *([a-z0-9@\-_\.]+)/gi) || [])[0];
+
+            if (token && user) {
+                const npmConfig: string = execSync('npm config ls -l').toString();
+                const npmrcPath: string | undefined = (npmConfig.match(/userconfig *= *".*"/gi) || [])[0];
+                const cleanToken: string = token.replace(/repo\.cplace\.apiToken: */, '');
+                const cleanUser: string = user.replace(/repo\.cplace\.apiTokenUser: */, '');
+                const cleanNpmrcPath: string = npmrcPath.replace(/userconfig *= */, '').replace(/"/gi, '');
+                if (!cleanNpmrcPath) {
+                    console.error('✗ No user npmrc found');
+                } else {
+                    const currentNpmrcConfig: string = fs.readFileSync(cleanNpmrcPath).toString();
+                    const isConfigured: boolean = currentNpmrcConfig.includes(registry);
+                    if (!isConfigured) {
+                        this.writeNPMRC(registry, cleanNpmrcPath, Buffer.from(`${cleanUser}:${cleanToken}`).toString('base64'), cleanUser);
+                    } else {
+                        console.info('⇢ cplace npmrc configuration for jfrog already found');
+                    }
+                }
+            } else {
+                console.error('✗ Gradle / jfrog credentials not found or configured correctly');
+            }
+        }
+    }
+
+    private writeNPMRC(registry, npmrcPath, auth, user) {
+        let npmrc = `@cplace-next:registry=https:${registry} \n`;
```

Comment:
How this works in watch mode?

Reply:
Only writes if it hasn't done so previously